pkg/kademlia/kademlia.go

```diff
@@ -275,16 +277,23 @@ func (k *Kademlia) FetchInfo(ctx context.Context, node pb.Node) (_ *pb.InfoRespo
 	return info, nil
 }
 
-// FindNode looks up the provided NodeID first in the local Node, and if it is not found
-// begins searching the network for the NodeID. Returns and error if node was not found
+// FindNode wraps a call to FindNodeWithVouchers with no vouchers
 func (k *Kademlia) FindNode(ctx context.Context, nodeID storj.NodeID) (_ pb.Node, err error) {
+	defer mon.Task()(&ctx)(&err)
+	return k.FindNodeWithVouchers(ctx, nodeID, nil, nil)
```

Comment:
do you have to check if 'nodeID' is not nil before calling 
`k.FindNodeWithVouchers(ctx, nodeID, nil, nil)`

Reply:
self is checked here
```
func (dialer *Dialer) Lookup(ctx context.Context, self *pb.Node, ask pb.Node, find storj.NodeID, limit int, vouchers []*pb.Voucher) (_ []*pb.Node, err error) {
	...
	if self != nil {
		req.Pingback = true
		req.Sender = self	
		req.Vouchers = vouchers
	}
        ...
}
```